modules/face/include/opencv2/face/face_alignment.hpp

```diff
@@ -0,0 +1,307 @@
+        *@param modelFilename stores the model after training
+        *@returns A boolean value
+        */
+        virtual bool train(std::vector<Mat>& images, std::vector< std::vector<Point2f> >& landmarks,std::string configfile,std::string modelFilename= std::string("Facemark_Kazemi.dat"))=0;
+        /** @brief This function is used to load the stored model.
+        *@param filename stores the name of the file containing the trained model
+        */
+        virtual bool load(std::string filename)=0;
+        /** @brief This function returns the landmarks of the face
+        *@param image stores the image in which landmarks have to be detected
+        *@param faces stores the bounding rectangle of faces detected in an image
+        *@param shapes stores the landmarks of each face in the image
+        */
+        virtual bool getShape(Mat image,std::vector<Rect> faces, std::vector< std::vector<Point2f> >& shapes)=0;
+        virtual ~FacemarkKazemi();
+};
+CV_EXPORTS Ptr<FacemarkKazemi> createFacemarkKazemi(CascadeClassifier face_cascade);
+/**@brief structure determining split in regression tree
+*/
+struct splitr{
+        //!index1 Index of the first coordinates among the test coordinates for deciding split.
+        long index1;
+        //! index2 index of the second coordinate among the test coordinates for deciding split.
+        long index2;
+        //! thresh threshold for deciding the split.
+        float thresh;
+};
+/** @brief represents a node of the regression tree*/
+struct node_info{
+    //First pixel coordinate of split
+    long index1;
+    //Second pixel coordinate .split
+    long index2;
+    long depth;
+    long node_no;
+};
+/** @brief regression tree structure. Each leaf node is a vector storing residual shape.
+* The tree is represented as vector of leaves.
+*/
+struct tree_node{
+    splitr split;
+    std::vector<Point2f> leaf;
+};
+struct regtree{
+    std::vector<tree_node> nodes;
+};
+/** @brief Represents a training sample
+*It contains current shape, difference between actual shape
+*and current shape. It also stores the image whose shape is being
+*detected.
+*/
+struct training_sample{
+    //! shapeResiduals vector which stores the residual shape remaining to be corrected.
+    std::vector<Point2f> shapeResiduals;
+    //! current_shape vector containing current estimate of the shape
+    std::vector<Point2f> current_shape;
+    //! actual_shape vector containing the actual shape of the face or the ground truth.
+    std::vector<Point2f> actual_shape;
+    //! image A mat object which stores the image.
+    Mat image ;
+    //! pixel_intensities vector containing pixel intensities of the coordinates chosen for testing
+    std::vector<int> pixel_intensities;
+    //! pixel_coordinates vector containing pixel coordinates used for testing
+    std::vector<Point2f> pixel_coordinates;
+};
+class FacemarkKazemiImpl : public face::FacemarkKazemi{
```

Comment:
"Impl" class should be moved into "src" directory (into private header `.hpp` file)